crates/pallet-subspace/src/lib.rs

```diff
@@ -788,7 +814,22 @@ impl<T: Config> Pallet<T> {
         CurrentSlot::<T>::put(pre_digest.slot);
 
         {
-            let key = (pre_digest.solution.public_key.clone(), pre_digest.slot);
+            let farmer_public_key = pre_digest.solution.public_key.clone();
+
+            // Optional restriction for block authoring to the root user
+            if !AllowAuthoringByAnyone::<T>::get() {
+                RootPlotPublicKey::<T>::mutate(|maybe_root_plot_public_key| {
```

Comment:
`mutate` threw me off a bit since this gets called on every block as we are trying to mutate everytime instead of just the first time

Reply:
Well, we need to read it either way (and it shouldn't update unless the value has changed), so should be fine as is I think.